riscv-page-tables/src/page_table.rs

```diff
@@ -1207,6 +1269,137 @@ impl<T: PagingMode> GuestStagePageTable<T> {
         }))
     }
 
+    /// Verifies the entire virtual address range is invalidated and that each page is one size
+    /// smaller than the requested page size. Also checks the pages are physically contiguous and
+    /// suitably aligned to the requested page size. It's important that all the pages must be in
+    /// the same state, either "Blocked" and uniquely owned by the TVM, or "BlockedShared" and not
+    /// owned by the TVM. Last thing to check is that each page must have been invalidated with a
+    /// TLB version that's older than the current one.
+    pub fn promote_page(
+        &self,
+        vaddr: PageAddr<T::MappedAddressSpace>,
+        requested_page_size: PageSize,
+        expected_page_size: PageSize,
```

Comment:
i'd only pass one of these in in since one can be inferred from the other

Reply:
Ok I'll do the calculation directly inside this function.